frontend/src/scenes/plugins/pluginActivityDescriptions.tsx

```diff
@@ -169,5 +169,48 @@ export function pluginActivityDescriber(logItem: ActivityLogItem): HumanizedChan
         }
     }
 
+    if (logItem.activity.startsWith('attachment_')) {
+        for (const change of logItem.detail.changes || []) {
+            let changeWording: string | JSX.Element = ''
+
+            if (logItem.activity === 'attachment_created') {
+                changeWording = (
+                    <>
+                        attached a file <code>{change.after}</code>
+                    </>
+                )
+            } else if (logItem.activity == 'attachment_updated') {
+                if (change.after === change.before) {
+                    changeWording = (
+                        <>
+                            updated attached file <code>{change.after}</code>
+                        </>
+                    )
+                } else {
+                    changeWording = (
+                        <>
+                            updated attached file from <code>{change.before}</code> to <code>{change.after}</code>
+                        </>
+                    )
+                }
+            } else if (logItem.activity === 'attachment_deleted') {
+                changeWording = (
+                    <>
+                        deleted attached file <code>{change.before}</code>
+                    </>
+                )
+            }
+
+            return {
```

Comment:
This will display only the first change right? Can there be more than one here?

Reply:
correct, as I have the backend currently, the changes list will only ever contain 1 item. if N plugin attachments are updated in one go, it'll generate N activity log entries, each with one change.

in my testing I didn't see any extant plugins that supported multiple attachments, so it didn't seem necessary to special-case handling that (because that can get arbitrarily complex, eg add a file, delete another file, and change a third, all in one update)